src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    @Test
+    void registerUserWithNullLogin() {
+        User userWithLogin = new User(USER_ID1, (String) NULL_VALUE, PASSWORD, AGE2);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithLogin);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the null user login, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithNullPassword() {
+        User userWithPassword = new User(USER_ID1, LOGIN, (String) NULL_VALUE, AGE1);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithPassword);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the null user password, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithAllUnrealValues() {
+        User userWithLoginAndPasswordAndAge = new User(USER_ID1, (String) NULL_VALUE,
+                (String) NULL_VALUE, UNREAL_AGE);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithLoginAndPasswordAndAge);
+        }, "to be thrown for the null user login and password and unreal age, but it wasn't ");
+    }
+
+    @Test
+    void registerUserWithNullPasswordAndLogin() {
+        User userWithLoginAndPassword = new User(USER_ID1, (String) NULL_VALUE,
+                (String) NULL_VALUE, AGE2);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithLoginAndPassword);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the null user login and password, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithShotPassword() {
+        User userWithShortPassword = new User(USER_ID1, LOGIN, SHORT_PASSWORD, AGE2);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the short user password, but it wasn't");
+    }
+
+    @Test
+    void registerUserWithBadAge() {
+        User userWithBadAge = new User(USER_ID1, LOGIN, PASSWORD, SMALL_AGE);
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(userWithBadAge);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the small user age, but it wasn't");
+    }
+
+    @Test
+    void registerTheSameUsers() {
+        assertEquals(user1, storage.get(LOGIN1));
+
+        user2 = user1;
+        assertThrows(UserIsNotValidException.class, () -> {
+            registrationService.register(user2);
+        }, "Expected " + UserIsNotValidException.class.getName()
+                + "to be thrown for the same user login, but it wasn't");
+    }
```

Comment:
can we reformat it to the test like: `register_userWithAlreadyExistingLogin_throwsException` and how should we add this existing user?